### PR TITLE
feat(streams): add config.streams_path for decoupled SSE endpoint URL

### PR DESCRIPTION
**Pull request by mhenrixon**
## Summary

- Adds `config.streams_path` option (default: `nil`) that lets host apps override the URL emitted by `pgbus_stream_src`
- When the engine is mounted behind an auth constraint (e.g. `AdminConstraint` at `/admin/jobs`), the derived streams URL is unreachable for non-admin users — this config eliminates the need for monkey-patches
- Resolution order: `config.streams_path` > engine route helper > fallback `/pgbus/streams`

### Usage

```ruby
# config/routes.rb
authenticate :user, ->(u) { u.admin? } do
  mount Pgbus::Engine => "/admin/jobs"
end
mount Pgbus::Web::StreamApp.new => "/pgbus/streams"

# config/initializers/pgbus.rb
Pgbus.configure { |c| c.streams_path = "/pgbus/streams" }
```

## Test plan

- [x] Existing `streams_helper_spec.rb` tests pass (19 examples)
- [x] New test verifies `config.streams_path` bypasses engine mount detection
- [x] Default behavior (nil) unchanged — engine route helper still used

## Release Notes

* **New Features**
  * Introduced `streams_path` configuration option, allowing customization of the base URL path used by server-sent event (SSE) streams with intelligent fallback to default paths when unconfigured.

* **Tests**
  * Added test coverage for the new configuration option, verifying proper path resolution and default fallback behavior.

### CODE REVIEW

File: app/helpers/pgbus/streams_helper.rb [hunk 1]
```diff
@@ -75,22 +75,34 @@ def compute_since_id(stream_name, replay)
...
+    #        Pgbus.configure { |c| c.streams_path = "/pgbus/streams" }
+    #
+    #   2. Engine route helper — derives the path from wherever the
+    #      host app mounted the engine.
+    #
+    #   3. Fallback `/pgbus/streams` — test-only context where the
+    #      engine's url_helpers aren't wired in.
     def pgbus_stream_src(signed_name)
+      base = Pgbus.configuration.streams_path
+      return "#{base}/#{signed_name}" if base
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply app/helpers/pgbus/streams_helper.rb]: Fixed in e6c3689. Added `delete_suffix("/")` before concatenation and a test for the trailing-slash edge case.
